app/Platform/Commands/SyncGameSets.php

```diff
@@ -4,35 +4,120 @@
+                    ['type' => GameSetType::HUB, 'title' => $game->title, 'image_asset_path' => $game->ImageIcon],
+                );
+            } else {
+                GameSet::updateOrCreate(
+                    ['game_id' => $game->id],
+                    ['type' => GameSetType::SIMILAR_GAMES, 'title' => 'Similar Games', 'image_asset_path' => null],
+                );
+            }
+
+            $progressBar->advance();
+        }
+        $progressBar->finish();
+
+        $gameAltsCount = GameAlternative::count();
+        $this->info("\nPopulating {$gameAltsCount} game_set_games and game_set_links...");
 
-    // protected function preProcessEntity(object $origin, array $transformed): array
-    // {
-    // }
+        $progressBar = $this->output->createProgressBar($gameAltsCount);
 
-    // protected function postProcessEntity(object $origin, object $transformed): void
-    // {
-    // }
+        foreach (GameAlternative::cursor() as $gameAlt) {
+            $parentGame = Game::find($gameAlt->gameID);
+            $childGame = Game::find($gameAlt->gameIDAlt);
+
+            // Determine if a swap is needed to ensure the hub is always the parent.
+            if (
+                ($parentGame->ConsoleID !== System::Hubs && $childGame->ConsoleID === System::Hubs)
+                || ($gameAlt->gameID > $gameAlt->gameIDAlt)
+            ) {
+                // Swap to ensure the hub is the parent, or ensure consistent ordering.
+                $temp = $parentGame;
+                $parentGame = $childGame;
+                $childGame = $temp;
+            }
+
+            // Ensure the hub is always the parent in the relationship.
+            if ($childGame->ConsoleID === System::Hubs && $parentGame->ConsoleID !== System::Hubs) {
+                // Swap the parent and child so that the hub is always the parent.
+                $temp = $parentGame;
+                $parentGame = $childGame;
+                $childGame = $temp;
+            }
+
+            $parentGameSet = GameSet::firstWhere('game_id', $parentGame->id);
+            $childGameSet = GameSet::firstWhere('game_id', $childGame->id);
+
+            if ($parentGame->ConsoleID === System::Hubs && $childGame->ConsoleID === System::Hubs) {
+                GameSetLink::upsert([
+                    'parent_game_set_id' => $parentGameSet->id,
+                    'child_game_set_id' => $childGameSet->id,
+                    'created_at' => $childGame->Created,
+                    'updated_at' => $childGame->Updated,
+                ], uniqueBy: ['parent_game_set_id', 'child_game_set_id']);
+            } else {
+                GameSetGame::upsert([
+                    'game_set_id' => $parentGameSet->id,
+                    'game_id' => $childGame->id,
+                    'created_at' => $childGame->Created,
+                    'updated_at' => $childGame->Updated,
+                ], uniqueBy: ['game_set_id', 'game_id']);
```

Comment:
This doesn't seem to be copying the created_at/updated_at values correctly:
```
> select * from game_set_games where game_set_id=(select id
from game_sets where game_id=1471) order by game_id;
+--------+-------------+---------+---------------------+---------------------+------------+
| id     | game_set_id | game_id | created_at          | updated_at          | deleted_at |
+--------+-------------+---------+---------------------+---------------------+------------+
| 139861 |       28362 |    1405 | 2024-08-28 00:10:31 | 2024-08-28 00:10:31 | NULL       |
| 139931 |       28362 |    1449 | 2024-08-28 00:10:31 | 2024-08-28 00:10:31 | NULL       |
| 140055 |       28362 |    1470 | 2024-08-28 00:10:31 | 2024-08-28 00:10:31 | NULL       |
| 140062 |       28362 |    1742 | 2024-08-28 00:10:31 | 2024-08-28 00:10:31 | NULL       |
| 154348 |       28362 |    1796 | 2024-08-28 00:11:15 | 2024-08-28 00:11:15 | NULL       |
| 140064 |       28362 |    1818 | 2024-08-28 00:10:31 | 2024-08-28 00:10:31 | NULL       |
| 154352 |       28362 |    2050 | 2024-08-28 00:11:15 | 2024-08-28 00:11:15 | NULL       |
| 154354 |       28362 |    2051 | 2024-08-28 00:11:15 | 2024-08-28 00:11:15 | NULL       |
| 154356 |       28362 |    2052 | 2024-08-28 00:11:15 | 2024-08-28 00:11:15 | NULL       |
| 140066 |       28362 |    3818 | 2024-08-28 00:10:31 | 2024-08-28 00:10:31 | NULL       |
+--------+-------------+---------+---------------------+---------------------+------------+
```

Reply:
Managed to screw this up when I consolidated the code into the action. Fixed in latest (and used your provided query to verify).

Reply:
Good catch. In latest, I've updated that `$createdAt` definition in the action to:
```php
$createdAt = $existingGameAlt?->Created ?? $existingGameAlt?->Updated ?? Carbon::now();
```